src/publish_control_board_rev3.cpp

```diff
@@ -453,3 +464,40 @@ void PublishControlBoardRev3::publish_lights_horn_wipers_message(const sensor_ms
     wiper_cmd_pub.publish(wiper_cmd_pub_msg);
   }
 }
+
+void PublishControlBoardRev3::publish_door_signal_message(const sensor_msgs::Joy::ConstPtr& msg)
+{
+  pacmod_msgs::SystemCmdInt door_signal_cmd_pub_msg;
+
+  door_signal_cmd_pub_msg.enable = local_enable;
+  door_signal_cmd_pub_msg.ignore_overrides = false;
+
+  // If the enable flag just went to true, send an override clear
+  if (!prev_enable && local_enable)
+  {
+    door_signal_cmd_pub_msg.clear_override = true;
+    door_signal_cmd_pub_msg.clear_faults = true;
+  }
+
+  if (vehicle_type == JUPITER_SPIRIT)
+  {
+    if (controller != HRI_SAFE_REMOTE)
+    {
+      if (msg->axes[axes[DPAD_UD]] == AXES_MAX && msg->buttons[btns[LEFT_BUMPER]] == BUTTON_DOWN)
+        door_signal_cmd_pub_msg.command = SLIDING_DOOR_CLOSE;
+      else if (msg->axes[axes[DPAD_UD]] == AXES_MIN && msg->buttons[btns[LEFT_BUMPER]] == BUTTON_DOWN)
+        door_signal_cmd_pub_msg.command = SLIDING_DOOR_OPEN;
+      else if (local_enable != prev_enable)
+        door_signal_cmd_pub_msg.command = last_door_cmd;
+      else
+        door_signal_cmd_pub_msg.command = SLIDING_DOOR_NEUTRAL;
+
+      if (last_axes.empty() ||
```

Comment:
What does this do? can you add a comment to explain these next 5 lines?

Reply:
DoorSignalState is enum and like you said neutral state is needed in transition between opening and closing. And the lines after that are just about publishing the message only when enabled, or when the state changes between enabled/disabled or when button state is changed.

Reply:
Thanks! looks like the indenting of your added comment is off though.